mpas_analysis/ocean/meridional_heat_transport.py

```diff
@@ -0,0 +1,213 @@
+import os
+
+from ..shared.constants.constants import rad_to_deg, monthDictionary
+from ..shared.plot.plotting import plot_vertical_section,\
+    setup_colormap, plot_1D
+
+from ..shared.io.utility import build_config_full_path, make_directories
+
+from ..shared.generalized_reader.generalized_reader \
+    import open_multifile_dataset
+
+from ..shared.timekeeping.utility import get_simulation_start_time
+
+from ..shared.analysis_task import setup_task, check_analysis_enabled
+from ..shared.climatology import climatology
+
+
+def meridional_heat_transport(config):  # {{{
+    """
+    Process MHT analysis member data if available.
+    Plots MHT as:
+       1D function of latitude
+       2D function of latitude and depth
+
+    Parameters
+    ----------
+    config : instance of MpasAnalysisConfigParser
+        configuration options used to customize the analysis task
+
+    Authors
+    -------
+    Mark Petersen
+
+    Last Modified
+    -------------
+    04/21/2017
+    """
+
+    # **** Initial settings ****
+
+    # perform common setup for the task
+    namelist, runStreams, historyStreams, calendar, namelistMap, streamMap, \
+        variableMap, plotsDirectory = setup_task(config, componentName='ocean')
+
+    check_analysis_enabled(
+        namelist=namelist,
+        analysisOptionName='config_am_timeseriesstatsmonthly_enable',
+        namelistMap=namelistMap,
+        raiseException=True)
+    check_analysis_enabled(
+        namelist=namelist,
+        analysisOptionName='config_am_meridionalheattransport_enable',
+        namelistMap=namelistMap,
+        raiseException=False)
+
+    # Get a list of timeSeriesStats output files from the streams file,
+    # reading only those that are between the start and end dates
+    #   First a list necessary for theMHT climatology
+    streamName = historyStreams.find_stream(streamMap['timeSeriesStats'])
+    startDateClimo = config.get('climatology', 'startDate')
+    endDateClimo = config.get('climatology', 'endDate')
+    inputFilesClimo = historyStreams.readpath(streamName,
+                                              startDate=startDateClimo,
+                                              endDate=endDateClimo,
+                                              calendar=calendar)
+    simulationStartTime = get_simulation_start_time(runStreams)
+
+    print '\n  List of files for climatologies:\n' \
+          '    {} through\n    {}'.format(
+              os.path.basename(inputFilesClimo[0]),
+              os.path.basename(inputFilesClimo[-1]))
+
+    startYearClimo = config.getint('climatology', 'startYear')
+    endYearClimo = config.getint('climatology', 'endYear')
+    #   Create dictionary to store Climo related variables
+    dictClimo = {'inputFilesClimo': inputFilesClimo,
+                 'startDateClimo': startDateClimo,
+                 'endDateClimo':  endDateClimo,
+                 'startYearClimo': startYearClimo,
+                 'endYearClimo': endYearClimo}
```

Comment:
do we need to define these dictionaries? We had them in the MOC to reduce variable passing through function calls, but in this case they don't seem necessary.